packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,517 @@
+      ruleSearchStringIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      validateFqdn(props);
+      validateIpAddress(props);
+      break;
+    }
+    case HealthCheckType.HTTPS: {
+      ruleAlarmIdentifierIsNotAllowed(props);
+      ruleSearchStringIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      validateFqdn(props);
+      validateIpAddress(props);
+      break;
+    }
+    case HealthCheckType.HTTP_STR_MATCH: {
+      ruleAlarmIdentifierIsNotAllowed(props);
+      validateSearchStringForStringMatch(props);
+      ruleEnableSNIIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      validateFqdn(props);
+      validateIpAddress(props);
+      break;
+    }
+    case HealthCheckType.HTTPS_STR_MATCH: {
+      ruleAlarmIdentifierIsNotAllowed(props);
+      validateSearchStringForStringMatch(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      validateFqdn(props);
+      validateIpAddress(props);
+      break;
+    }
+    case HealthCheckType.TCP: {
+      ruleAlarmIdentifierIsNotAllowed(props);
+      ruleEnableSNIIsNotAllowed(props);
+      ruleSearchStringIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      validateFqdn(props);
+      validateIpAddress(props);
+      break;
+    }
+    case HealthCheckType.RECOVERY_CONTROL: {
+      ruleAlarmIdentifierIsNotAllowed(props);
+      ruleRoutingControlIsRequired(props);
+      ruleEnableSNIIsNotAllowed(props);
+      ruleSearchStringIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      rulePortIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      break;
+    }
+    case HealthCheckType.CALCULATED: {
+      validateChildHealthChecks(props);
+      ruleAlarmIdentifierIsNotAllowed(props);
+      rulePortIsNotAllowed(props);
+      ruleEnableSNIIsNotAllowed(props);
+      ruleSearchStringIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsRequired(props);
+      break;
+    }
+    case HealthCheckType.CLOUDWATCH_METRIC: {
+      ruleAlarmIdentifierIsRequired(props);
+      rulePortIsNotAllowed(props);
+      ruleEnableSNIIsNotAllowed(props);
+      ruleSearchStringIsNotAllowed(props);
+      ruleChildHealthChecksIsNotAllowed(props);
+      ruleRoutingControlIsNotAllowed(props);
+      ruleHealthThresholdIsNotAllowed(props);
+      break;
+    }
+    default:
+      throw new Error(`Unsupported health check type: ${props.type}`);
```

Comment:
Switching to a map would also remove the need for this exception, since the `rules` type will require that all possible `HealthCheckType` values are covered